sys-boot/refind/refind-0.11.4.ebuild

```diff
@@ -0,0 +1,187 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit flag-o-matic multiprocessing toolchain-funcs
+
+DESCRIPTION="The UEFI Boot Manager by Rod Smith"
+HOMEPAGE="https://www.rodsbooks.com/refind/"
+SRC_URI="mirror://sourceforge/project/${PN}/${PV}/${PN}-src-${PV}.tar.gz"
+
+LICENSE="BSD GPL-2 GPL-3 FDL-1.3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+FS_USE="btrfs +ext2 +ext4 hfs +iso9660 ntfs reiserfs"
+IUSE="${FS_USE} gnuefi doc custom-cflags"
+
+DEPEND="gnuefi? ( >=sys-boot/gnu-efi-3.0.2 )
+	!gnuefi? ( >=sys-boot/udk-2018-r1 )"
+
+DOCS=(NEWS.txt README.txt docs/refind docs/Styles)
+
+UDK_WORKSPACE="${WORKDIR}/udk"
+
+pkg_pretend() {
+	if use custom-cflags; then
+		ewarn
+		ewarn "You have enabled building with USE=custom-cflags. Be aware that"
+		ewarn "using this can result in EFI binaries that fail to run and may"
+		ewarn "fail to build at all. This is strongly advised against by upstream."
+		ewarn
+		ewarn "See https://bugs.gentoo.org/598587#c3 for more information"
+		ewarn
+	fi
+}
+
+pkg_setup() {
+	if use x86; then
+		export EFIARCH=ia32
+		export BUILDARCH=ia32
+	elif use amd64; then
+		export EFIARCH=x64
+		export BUILDARCH=x86_64
+	fi
+}
+
+src_prepare() {
+	default
+
+	# bug 598647 - PIE not supported
+	sed -e '/^CFLAGS/s:$: -fno-PIE:' -i Make.common || die
+
+	# Update Makefile
+	sed -e "s:^\(\s*\)\. \./edksetup\.sh[^&]*:\1build -a \$(UC_ARCH) -p MdePkg/MdePkg.dsc :" \
+		-i Makefile || die "Could not update Makefile"
+
+	# Prepare UDK workspace
+	if ! use gnuefi; then
+		mkdir "${UDK_WORKSPACE}" || die
+		ln -s "${EPREFIX}"/usr/share/udk/{Mde,IntelFramework}{,Module}Pkg \
+			"${UDK_WORKSPACE}" || die "Could not link UDK files"
+			
+		# HFS filesystem is not working with UDK
+		# See https://sourceforge.net/p/refind/discussion/general/thread/4e203f0a10/
+		sed -e "/RefindPkg\/filesystems\/hfs\.inf/d" -i RefindPkg.dsc || die
+		if use hfs; then
+			eerror "hfs use flag ignored because not working with UDK"
+		fi
+	fi
+}
+
+src_configure() {
+	if ! use gnuefi; then
+		. udk-workspace "${UDK_WORKSPACE}" || die
+		sed -e "s:^\(MAX_CONCURRENT_THREAD_NUMBER\s*=\).*$:\1 $(makeopts_jobs):" \
+			-i "${UDK_WORKSPACE}/Conf/target.txt" || die "Failed to configure target file"
+		sed -e "s:\(_\(CC\|ASM\|PP\|VFRPP\|ASLCC\|ASLPP\|DLINK\)_PATH\s*=\).*$:\1 $(tc-getCC):" \
+			-e "s:\(_ASLDLINK_PATH\s*=\).*$:\1 $(tc-getLD):" \
+			-e "s:\(_OBJCOPY_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+			-e "s:\(_RC_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+			-e "s:\(_SLINK_PATH\s*=\).*$:\1 $(tc-getAR):" \
+			-i "${UDK_WORKSPACE}/Conf/tools_def.txt" \
+			|| die "Failed to prepare tools definition file"
+	fi
+}
+
+src_compile() {
+	# Ensure environment is set
+	if ! use gnuefi; then
+		. udk-workspace "${UDK_WORKSPACE}" || die
```

Comment:
Same as above, don't source external files in ebuild environment. Move it to a subshell around `emake` call.